### PR TITLE
Fix stream writer flushing to be thread safe.

### PR DESCRIPTION
**Pull request by benbjohnson**
## Overview

This change fixes the raft streams so that Flush() is not called asynchronously while the snapshot is being written. Previously the `logWriter` would call `Flush()` while the snapshot was being written. Now it will not call flush until the snapshot is complete.

### CODE REVIEW

**Comment by toddboom** [comment]
:shipit: on green

File: messaging/handler.go [hunk 1]
```diff
@@ -231,7 +231,9 @@ func CopyFlush(dst io.Writer, src io.Reader) (written int64, err error) {
 			}
 
 			// Flush after write.
-			if dst, ok := dst.(http.Flusher); ok {
```
Comment (otoolep) [inline messaging/handler.go]: Why this change?

**Comment by otoolep** [comment]
+1, the write will now only take place is snapshotting is not.

File: messaging/handler.go (hunk 1 above)
Comment (benbjohnson) [inline messaging/handler.go]: This one wasn't necessary. I mostly just don't like using http.Flusher because Flush() is such a generic concept that tying it to http seems dumb. There is no functional change here. It's just aesthetics.
